packages/discovery-react-components/src/components/SearchFacets/components/FacetsGroups/SingleSelectFacetsGroup.tsx

```diff
@@ -63,39 +74,68 @@ export const SingleSelectFacetsGroup: FC<SingleSelectFacetsGroupProps> = ({
 
   const handleOnClick = (event: SyntheticEvent<HTMLInputElement>): void => {
     const target: HTMLInputElement = event.currentTarget;
-    const name = target.getAttribute('name') || '';
-    const text = target.getAttribute('value') || '';
-    onChange(name, text, true);
-  };
+    const selectedFacetName = target.getAttribute('data-name') || '';
+    const selectedFacetKey = target.getAttribute('data-key') || '';
+    const checked = true;
 
-  const getLabel = (facetText: string, count: number | undefined) => {
-    return count !== undefined && showMatchingResults
-      ? formatMessage(messages.labelTextWithCount, { facetText: facetText, count: count }, false)
-      : formatMessage(messages.labelText, { facetText: facetText }, false);
+    // If this is in the Show more modal, we want to update tempSelectedFacets only
+    if (tempSelectedFacets && setTempSelectedFacets) {
+      setTempSelectedFacets([{ selectedFacetName, selectedFacetKey, checked }]);
+    } else {
+      // If this isn't in the Show more modal, we want to save the selection
+      onChange([{ selectedFacetName, selectedFacetKey, checked }]);
+    }
   };
 
+  // If this is in the Show more modal, we need to check for a temporary selection
+  // and add a suffix to the selected facet value to make it unique
+  let facetValueSelected = selectedFacet;
```

Comment:
It's mutating `selectedFacet` props.

Reply:
I don't think it is since `selectedFacet` is a string.